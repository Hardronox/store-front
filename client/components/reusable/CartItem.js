import React from 'react';
import DescriptionItem from './DescriptionItem';
import NumberFormat from 'react-number-format';

const CartItem = (props) => {
  const {item} = props;

  const renderDescription = () => {
    return Object.entries(item.description).map(([k, v], i) => {
      return <DescriptionItem title={k} value={v} key={i} />;
    });
  };
  return (
    <div className="cart-item flex-col">
      <div className="cart-item-header flex-row space-between">
        <div className="item-seller">
                    Seller:
          <span>
            {` ${item.seller}`}
          </span>

        </div>
        <div className="flex-row centered">
          <div className="item-header-buttons">
            <button className="item-header-button">
                            Pay only this seller
            </button>
            <button className="item-header-button" disabled>
                            Request total from seller
            </button>
          </div>
          <div className="item-header-info">
            <i className="fa fa-info-circle" aria-hidden="true"></i>
          </div>
        </div>

      </div>
      <div className="cart-item-body flex-row">
        <div className="item-image-container flex-col">
          <img src={item.image} alt="item image" className="item-image"/>
        </div>
        <div className="item-description-container flex-col">
          <div className="item-title">
            {item.title}
          </div>
          <div className="item-description flex-col">
            {renderDescription()}
          </div>
        </div>
        <div className="item-price-container flex-col">
          <div className="flex-row">
            <div className="item-quantity">
                            Quantity: {item.quantity}
            </div>
            <div className="item-price">
              <NumberFormat value={item.price} displayType={'text'} thousandSeparator={true} prefix={'US $'} />
            </div>
          </div>

          <div className="flex-row">
            <div className="item-shipping flex-col">
                            Expedited
              <span>Expedited International Shipping</span>
            </div>
            <div className="item-shipping-price">
              <NumberFormat value={item.shippingCharge} displayType={'text'} thousandSeparator={true} prefix={'+US $'} />
            </div>
          </div>

        </div>
      </div>
      <div className="cart-item-footer flex-row">
        <div className="cart-item-buttons">
          <button className="btn btn-link" onClick={() => props.onRemove(item.id)}>Remove</button>
          <button className="btn btn-link">Save for later</button>
        </div>
      </div>
    </div>
  );
};

export default CartItem;
