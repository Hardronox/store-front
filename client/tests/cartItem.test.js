import React from 'react';
import CartItem from '../components/reusable/CartItem';
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
configure({
    adapter: new Adapter()
});

const mockItem = {
    id: 6,
    seller: "acronwing23342",
    image: "https://iheartcats.com/wp-content/uploads/2015/02/5255462794_19f91d3463_z-e1424134684199.jpg",
    title: "Kitty",
    description: {
    "condition": "battered",
        "color": "gray",
        "model": "iKitty 6S",
        "storage": "64GB"
        },
    price: 121.99
}

describe("<CartItem/>", () => {
    const wrapper = shallow(<CartItem item={mockItem} />)
    it("should contain 4 button elements", () => {
        expect(wrapper.find('button').length).toBe(4)
    })
    it("calls removeItem Redux action creator on button click", () => {
        const props = {
            onRemove: jest.fn(),
            item: mockItem
        }
        const wrapper = shallow(<CartItem {...props} />)
        wrapper.find('button').at(2).simulate('click')
        expect(props.onRemove).toHaveBeenCalledWith(6)
    })
    it("renders correctly", () => {
        const tree = shallow*<CartItem item={mockItem}/>
        expect(toJson(tree)).toMatchSnapshot()
    })
});

// it("calls ComponentDidMount 0 times", () => {
//     jest.spyOn(CartItem.prototype.componentDidMount)
//     expect(CartItem.prototype.componentDidMount.mock.calls.length).toBe(0)
// })