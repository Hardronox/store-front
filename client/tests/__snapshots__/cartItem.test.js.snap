// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartItem/> renders correctly 1`] = `undefined`;
